### ISSUE TITLE
Fermentables wrong colour unit in default database

### ISSUE THREAD

**Issue by jensnielsen** [Brewtarget/brewtarget#366]
Hi

I noticed a lot of my fermentables had their colour listed in the wrong unit so I scrapped my database and restored the default one (brewtarget/data/default_db.sqlite from develop branch) but I still have the same issue. I haven't checked them all but those I have checked are wrong. For example

- "Caramel/Crystal Malt - 80L" is listed as 80 srm but based on the name it should be an 80 Lovibond malt
- "Weyermann - Carafa I" is listed as 350 srm but according to Weyermann it's a 350 Lovibond malt (https://www.weyermann.de/eng/produkte.asp?idkat=19&umenue=yes&idmenue=&sprache=2).

And so on. The numbers are in Lovibond but shown and used as srm.

Or am I missing something? I'm on 2.3.0 for Windows.

**Comment by kapinga** [comment]
As far as I can tell, Lovibond is approximately equivalent to SRM (see [BeerSmith] (http://beersmith.com/blog/2008/04/29/beer-color-understanding-srm-lovibond-and-ebc/) and [Wikipedia] (https://en.wikipedia.org/wiki/Beer_measurement#Colour)).

**Comment by jensnielsen** [comment]
Nice, that's interesting. I based my knowledge of colours on this page, which has quite a big difference between SRM and L: https://www.brewtoad.com/tools/color-converter

The reason I discovered this is because I added a recipe for a clone beer and didn't get the colour the recipe told me I'd get (according to the slider in Brewtarget, I haven't actually checked the colour of a finished product) but if I change the colour values of the malts according to the Brewtoad formula it's spot on.

But I guess we can agree Lovibond is obsolete and should not be used then

**Comment by kapinga** [comment]
**tl;dr:** Brewtarget is actually showing the Lovibond color of the malts/fermentables, even though it says SRM. The color calculations the correctly use these Lovibond color units to estimate the final beer color in SRM.

---

I answered too quickly last night, the situation is somewhat more complex than I thought. This [StackExchange](http://homebrew.stackexchange.com/questions/2632/how-do-you-convert-between-srm-and-lovibond) thread has a good discussion. Lovibond is still commonly used to measure malt color as you noted (at least in the US, I can't speak to elsewhere), but SRM (which relies on transmission through a liquid) is used for beer color. There are several formulas for estimating the final beer's SRM color based on the color of the malt, although the appropriate units for malt color aren't entirely clear.

Brewtarget takes all this into account (you can select which formula is used for calculating color in the options dialog). The [BeerXML format](http://www.beerxml.com/beerxml.htm) (which Brewtarget is based on internally, for better or for worse) even notes that the color of malts should be recorded in Lovibond degrees (but SRM for liquid extract).

Without worrying about units for the moment, Brewtarget calculates the beer's color by adding up all the 'Malt Color Unit' (MCU) contribution of each fermentable ([MCU = grain_color * lbs_grain / gal_water](http://www.brewwiki.com/index.php/Estimating_Color)). The MCU value is then fed into whichever formula the user has chosen (by default [Morey's approximation](https://www.morebeer.com/brewingtechniques/beerslaw/morey.html)) to predict the SRM of the final beer. Morey notably did not 

The question then is: what units are appropriate for calculating MCU? Brewtarget internally acts as if all fermentables have SRM color units, but I'm having trouble figuring out what color units should appropriately be used when calculating MCU. Although SRM and Lovibond are supposedly 'similar', [Wikipedia](https://en.wikipedia.org/wiki/Standard_Reference_Method#Measurement_method) later claims that the formula to convert them is:
> SRM = 1.3546 x L - 0.76
This is similar for small numbers (5 L == 6 SRM), but definitely introduces significant differences at high levels of coloration. As far as I can tell, most calculations implicitly use the Lovibond color of the malt to get the MCU, so that seems like the correct calculation is:
> Lovibond malt color -> MCU -> SRM beer color

Assuming that should be MCU calcuated using Lovibond units, then Brewtarget is doing the calculations correctly. The only problem then is that Brewtarget says that malt colors are in SRM when they are actually in degrees Lovibond. (There's a secondary problem that (liquid) extracts should probably be treated differently, but aren't as far as I can tell.)

**Comment by pricelessbrewing** [comment]
@kapinga I think until we have more information on what MCU is supposed to use, some brief research shows it should probably be lovibond used in the fermentable database and the unit labels changed accordingly. I think it's better to be up front, and be wrong (potentially), then to misslabel the units we're using. 

fermentables: lovibond

Beer color: srm.

**Comment by pricelessbrewing** [comment]
Additionally, malsters always provide their color information in lovibond, not srm.

**Comment by matty0ung** [comment]
As of the change to `Unit.cpp` in https://github.com/Brewtarget/brewtarget/pull/989, we have a corrected conversion between Lovibond and SRM.  Apologies for the time it has taken to get to this.

I will endeavour to look at the other points here at some point, hopefully sooner.

**Comment by matty0ung** [comment]
I realised that the partial fix I had done would be worse than no fix at all!  So I've done the rest of it.  Will be in the next release.

**Comment by matty0ung** [comment]
Fixed in [4.1.3](https://github.com/Brewtarget/brewtarget/releases/tag/v4.1.3).  As ever, please give us a shout if you hit any issues.
